src/backend/distributed/planner/relation_restriction_equivalence.c

```diff
@@ -734,6 +783,144 @@ AddToAttributeEquivalenceClass(AttributeEquivalenceClass **attributeEquivalanceC
 
 
 /*
+ * AddUnionSetOperationsToAttributeEquivalenceClass recursively iterates on all the
+ * setOperations and adds each corresponding target entry to the given equivalence
+ * class.
+ */
+static void
+AddUnionSetOperationsToAttributeEquivalenceClass(Query *query,
+												 SetOperationStmt *setOperation,
+												 Var *varToBeAdded,
+												 AttributeEquivalenceClass **
+												 attributeEquivalenceClass)
+{
+	Node *leftArgument = setOperation->larg;
+	Node *rightArgument = setOperation->rarg;
+
+	/* we're only interested in UNIONs */
+	if (setOperation->op != SETOP_UNION)
```

Comment:
> we already check this inside ErrorIfUnsupportedUnionQuery()

I realized that we call this function prior to `ErrorIfUnsupportedUnionQuery ()` to give more meaningful error message. So, we need to keep it as it is.